server/src/main/java/com/epam/aidial/core/server/security/IdentityProvider.java

```diff
@@ -153,6 +157,9 @@ private List<String> extractUserRoles(Map<String, Object> map) {
                 if (next instanceof List) {
                     return (List<String>) next;
                 } else if (next instanceof String) {
+                    if (rolesDelimiter != null) {
+                        return List.of(((String) next).split(rolesDelimiter));
```

Comment:
Maybe it makes sense to trim split role values. For example, when the delimiter is "," and the roles string contains a whitespace: "r1, r2". The result will be "r1", " r2". The second role value contains a leading whitespace.

Reply:
btw, if the roles string is "r1  r2  r3" (two whitespaces) and the delimiter is " " (single whitespace), then the roles list will contain 5 values (2 of them are empty strings)

Reply:
First example: your delimiter should be ", " then.
Second: that is true. However that shouldn't be the case with JWT tokens, I barely can't imagine IDP that issues roles with unpredictable number of delimiters between them. 
Anyway, let's maybe filter empty strings from the list of extracted roles. 
Thank you for that case.